Examples/Framework/include/ActsExamples/EventData/Measurement.hpp

```diff
@@ -28,253 +29,357 @@
+  using VariableProxy = VariableMeasurementProxy<Acts::eBoundSize, false>;
+  using ConstVariableProxy = VariableMeasurementProxy<Acts::eBoundSize, true>;
+
+  MeasurementContainer();
+
+  std::size_t size() const;
+
+  void reserve(std::size_t size);
+
+  std::size_t addMeasurement(std::uint8_t size);
+
+  VariableProxy getMeasurement(std::size_t index);
+  ConstVariableProxy getMeasurement(std::size_t index) const;
+
+  template <std::size_t Size>
+  FixedProxy<Size> getMeasurement(std::size_t index) {
+    return FixedProxy<Size>{*this, index};
+  }
+  template <std::size_t Size>
+  ConstFixedProxy<Size> getMeasurement(std::size_t index) const {
+    return ConstFixedProxy<Size>{*this, index};
   }
-  /// A measurement can only be constructed with valid parameters.
-  VariableSizeMeasurement() = delete;
-  VariableSizeMeasurement(const VariableSizeMeasurement&) = default;
-  VariableSizeMeasurement(VariableSizeMeasurement&&) = default;
-  ~VariableSizeMeasurement() = default;
-  VariableSizeMeasurement& operator=(const VariableSizeMeasurement&) = default;
-  VariableSizeMeasurement& operator=(VariableSizeMeasurement&&) = default;
 
-  /// Source link that connects to the underlying detector readout.
-  const Acts::SourceLink& sourceLink() const { return m_source; }
+  VariableProxy makeMeasurement(std::uint8_t size);
+  template <std::size_t Size>
+  FixedProxy<Size> makeMeasurement() {
+    return getMeasurement<Size>(addMeasurement(Size));
+  }
+
+  template <bool Const>
+  class IteratorImpl {
+   public:
+    using value_type =
+        std::conditional_t<Const, ConstVariableProxy, VariableProxy>;
+    using reference = value_type;
+    using pointer = value_type*;
+    using difference_type = std::ptrdiff_t;
+    using iterator_category = std::forward_iterator_tag;
+
+    using Container = std::conditional_t<Const, const MeasurementContainer,
+                                         MeasurementContainer>;
+
+    IteratorImpl(Container& container, std::size_t index)
+        : m_container(container), m_index(index) {}
+
+    reference operator*() const {
+      if constexpr (Const) {
+        return m_container.getMeasurement(m_index);
+      } else {
+        return m_container.getMeasurement(m_index);
+      }
+    }
+
+    pointer operator->() const { return &operator*(); }
+
+    IteratorImpl& operator++() {
+      ++m_index;
+      return *this;
+    }
+
+    IteratorImpl operator++(int) {
+      auto copy = *this;
+      ++*this;
+      return copy;
+    }
+
+    bool operator==(const IteratorImpl& other) const {
+      return m_index == other.m_index;
+    }
+
+    bool operator!=(const IteratorImpl& other) const {
+      return !(*this == other);
+    }
+
+   private:
+    Container& m_container;
+    std::size_t m_index;
+  };
+
+  using Iterator = IteratorImpl<false>;
+  using ConstIterator = IteratorImpl<true>;
+
+  Iterator begin();
+  Iterator end();
+  ConstIterator begin() const;
```

Comment:
I think STL convention is to call those `cbegin` and `cend`

Reply:
apparently one need `begin() const` to make range-based for loop to work. but I added `cbegin() const` as well